tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -0,0 +1,317 @@
+};
+
+template <bool HasPrimitiveVars, typename BoundaryCorrection>
+using get_correction_primitive_vars =
+    typename get_correction_primitive_vars_impl<HasPrimitiveVars>::template f<
+        BoundaryCorrection>;
+
+template <bool HasPrimitiveVars = false>
+struct get_system_primitive_vars_impl {
+  template <typename System>
+  using f = tmpl::list<>;
+};
+
+template <>
+struct get_system_primitive_vars_impl<true> {
+  template <typename System>
+  using f = typename System::primitive_variables_tag::tags_list;
+};
+
+template <bool HasPrimitiveVars, typename System>
+using get_system_primitive_vars = typename get_system_primitive_vars_impl<
+    HasPrimitiveVars>::template f<System>;
+
+template <typename BoundaryCorrection, typename... PackageTags,
+          typename... FaceTags, typename... VolumeTags, size_t Dim>
+void call_dg_package_data(
+    const gsl::not_null<Variables<tmpl::list<PackageTags...>>*> package_data,
+    const BoundaryCorrection& correction,
+    const Variables<tmpl::list<FaceTags...>>& face_variables,
+    const tuples::TaggedTuple<VolumeTags...>& volume_data,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& unit_normal_covector,
+    const boost::optional<tnsr::I<DataVector, Dim, Frame::Inertial>>&
+        mesh_velocity) {
+  boost::optional<Scalar<DataVector>> normal_dot_mesh_velocity{};
+  if (static_cast<bool>(mesh_velocity)) {
+    normal_dot_mesh_velocity =
+        dot_product(*mesh_velocity, unit_normal_covector);
+  }
+  correction.dg_package_data(
+      make_not_null(&get<PackageTags>(*package_data))...,
+      get<FaceTags>(face_variables)..., unit_normal_covector, mesh_velocity,
+      normal_dot_mesh_velocity, get<VolumeTags>(volume_data)...);
+}
+
+template <typename BoundaryCorrection, typename... BoundaryCorrectionTags,
+          typename... PackageTags>
+void call_dg_boundary_terms(
+    const gsl::not_null<Variables<tmpl::list<BoundaryCorrectionTags...>>*>
+        boundary_corrections,
+    const BoundaryCorrection& correction,
+    const Variables<tmpl::list<PackageTags...>>& interior_package_data,
+    const Variables<tmpl::list<PackageTags...>>& exterior_package_data,
+    const ::dg::Formulation dg_formulation) {
+  correction.dg_boundary_terms(
+      make_not_null(&get<BoundaryCorrectionTags>(*boundary_corrections))...,
+      get<PackageTags>(interior_package_data)...,
+      get<PackageTags>(exterior_package_data)..., dg_formulation);
+}
+
+template <typename System, typename BoundaryCorrection, size_t FaceDim,
+          typename... VolumeTags>
+void test_boundary_correction_impl(
+    const BoundaryCorrection& correction, const Mesh<FaceDim>& face_mesh,
+    const tuples::TaggedTuple<VolumeTags...>& volume_data,
+    const bool use_moving_mesh, const ::dg::Formulation dg_formulation,
+    const ZeroOnSmoothSolution zero_on_smooth_solution) {
+  CAPTURE(use_moving_mesh);
+  CAPTURE(dg_formulation);
+  CAPTURE(FaceDim);
+  using variables_tag = typename System::variables_tag;
```

Comment:
[optional] If I'm not mistaken, you only use this once on the line below, so you could remove this `using` by writing `System::variables_tag::tags_list` on the line below.